### PR TITLE
Created research export rake

### PR DESCRIPTION
**Pull request by ajparsons**
Currently is dumping csvs files in 'exports'  - there might be a better location than that.

Connects to #3734

### CODE REVIEW

**Comment by lizconlan** [comment]
@ajparsons would you mind re-running the rake task to check I haven't broken anything? (I'll fix it if I have!)

**Comment by ajparsons** [comment]
@lizconlan All changes seem fine (and are neater) - still works fine on test data. Any way of testing it on live data? Test data isn't good at testing the scale of the export.

**Comment by lizconlan** [comment]
@ajparsons foolish question (to which I could probably find the answer some other way) has it been run against live data before? (also, I can almost certainly run it against live data to see what happens)

File: lib/tasks/export.rake [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+# Exports a model
+#
+# query    - a query used to limit the export to matching records
+# header   - used to restrict exported columns
+# override - pass in lambdas to modify a given column based on values in the row
+#
+# Returns a String
+def csv_export(model, query=nil, header=nil, override={})
+  # set query and header to default values unless supplied
+  query = model.all unless query
```
Comment (garethrees) [inline lib/tasks/export.rake]: @lizconlan probably want to use `find_each` here rather than trying to load everything in to memory all at once?

Comment (lizconlan) [inline lib/tasks/export.rake]: @garethrees good point - thank you!

Comment (lizconlan) [inline lib/tasks/export.rake]: oh, wait - this breaks for `InfoRequestBatchPublicBody` as it doesn't have a primary key

File: lib/tasks/export.rake [hunk 2]
```diff
@@ -59,7 +75,7 @@ end
 # Returns a String
 def csv_export(model, query=nil, header=nil, override={})
   # set query and header to default values unless supplied
-  query = model.all unless query
+  query = model unless query
```
Comment (garethrees) [inline lib/tasks/export.rake]: Could do:

```ruby
query ||= model
```

Comment (lizconlan) [inline lib/tasks/export.rake]: ah, yes - much neater!

Comment (lizconlan) [inline lib/tasks/export.rake]: ...I'll apply the same to the next line as well

**Comment by ajparsons** [comment]
FYI I added a gender detector because it seemed better to do that at this point. I added a gem to do this, this change anything process-wise?

**Comment by lizconlan** [comment]
> I added a gem to do this, this change anything process-wise?

I think we've decided to tentatively allow it and review if it complicates things

**Comment by lizconlan** [comment]
ToDo:
 - [ ] squash/rebase
 - [ ] add date limits to queries to make sure it's not open-ended

**Pull request merged**